src/CoreBundle/Service/MaintenanceModeService.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace ChameleonSystem\CoreBundle\Service;
+
+use ChameleonSystem\CoreBundle\Exception\MaintenanceModeErrorException;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+
+class MaintenanceModeService implements MaintenanceModeServiceInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function isActivated(): bool
+    {
+        return file_exists(PATH_MAINTENANCE_MODE_MARKER);
+    }
+
+    public function isActivatedInDb(): bool
+    {
+        $col = $this->connection->fetchColumn("SELECT `shutdown_websites` FROM `cms_config`");
+
+        if (false === $col) {
+            return false;
+        }
+
+        return '1' === $col['shutdown_websites'];
+    }
+
+    public function activate(): void
+    {
+        try {
+            $this->connection->executeUpdate("UPDATE `cms_config` SET `shutdown_websites` = '1'");
```

Comment:
We need to invalidate the cache for cms_config afterwards.

Reply:
ok